src/main/java/net/neoforged/neoforge/network/handling/IPayloadContext.java

```diff
@@ -6,32 +6,77 @@
 package net.neoforged.neoforge.network.handling;
 
 import io.netty.channel.ChannelHandlerContext;
-import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Supplier;
+import net.minecraft.network.Connection;
 import net.minecraft.network.ConnectionProtocol;
 import net.minecraft.network.ProtocolInfo;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.protocol.Packet;
 import net.minecraft.network.protocol.PacketFlow;
-import net.minecraft.world.entity.Entity;
+import net.minecraft.network.protocol.common.custom.CustomPacketPayload;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.server.network.ConfigurationTask;
 import net.minecraft.world.entity.player.Player;
-import net.minecraft.world.level.Level;
 
 /**
- * Defines a phase-less payload context that is passed to a handler for a payload that arrives during the connection.
+ * Common context interface for payload handlers.
  */
 public interface IPayloadContext {
```

Comment:
I don't think there's any functionality that can be accomplished with the packet listener.  If needed, it can be retrieved from the player (the subclasses have public access to the strongly-typed listeners).

Reply:
Resolved to add it later if necessary; at face value it does not provide any useful functionality.